spacy/language.py

```diff
@@ -1429,25 +1473,37 @@ def from_config(
         # then we would load them twice at runtime: once when we make from config,
         # and then again when we load from disk.
         pipeline = config.get("components", {})
+        # If components are loaded from a source (existing models), we cache
+        # them here so they're only loaded once
+        source_nlps = {}
         for pipe_name in config["nlp"]["pipeline"]:
             if pipe_name not in pipeline:
                 opts = ", ".join(pipeline.keys())
                 raise ValueError(Errors.E956.format(name=pipe_name, opts=opts))
             pipe_cfg = util.copy_config(pipeline[pipe_name])
             if pipe_name not in disable:
-                if "factory" not in pipe_cfg:
+                if "factory" not in pipe_cfg and "source" not in pipe_cfg:
                     err = Errors.E984.format(name=pipe_name, config=pipe_cfg)
                     raise ValueError(err)
-                factory = pipe_cfg.pop("factory")
-                # The pipe name (key in the config) here is the unique name of the
-                # component, not necessarily the factory
-                nlp.add_pipe(
-                    factory,
-                    name=pipe_name,
-                    config=pipe_cfg,
-                    overrides=pipe_overrides,
-                    validate=validate,
-                )
+                if "factory" in pipe_cfg:
+                    factory = pipe_cfg.pop("factory")
+                    # The pipe name (key in the config) here is the unique name
+                    # of the component, not necessarily the factory
+                    nlp.add_pipe(
+                        factory,
+                        name=pipe_name,
+                        config=pipe_cfg,
+                        overrides=pipe_overrides,
+                        validate=validate,
+                    )
+                else:
+                    model = pipe_cfg["source"]
+                    if model not in source_nlps:
+                        # We only need the components here, no other data
+                        disable = ["vocab", "tokenizer"]
+                        source_nlps[model] = util.load_model(model, disable=disable)
```

Comment:
Did we remove the `vocab` keyword argument to `load_model`?

I think we'll have problems if `source_nlps[model].vocab` isn't the same instance as `self.vocab`. We've been trying to relax the requirement that the `Vocab` instance is the same across the pipeline, but we haven't really tested it, and I can imagine problems occurring.

Reply:
I think this falls more into the category of validation? Because if there are underlying problems that make the vocabs incompatible, it's not going to be solved by initializing the source models with the destination vocab... But I don't have a good answer for how to do this beyond (maybe) warning if the language is different.

Reply:
Hmm I do think it's an implementation issue rather than validation.

For instance, we might have a `morphologizer` component that expects to write to `self.vocab.morphology` during `begin_training`. It then expects to save out that state and load it back in `from_disk`. But if that `morphologizer` component has been sourced, its vocab instance won't be the one in the `nlp` object, so that information won't come back correctly.